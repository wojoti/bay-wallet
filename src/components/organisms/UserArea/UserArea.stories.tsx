import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import UserArea from './UserArea';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Organisms/UserArea',
  component: UserArea,
  argTypes: {
    onNotificationPress: {action: 'onNotificationPress'},
    onUserProfilePress: {action: 'onUserProfilePress'},
    onDetailsPress: {action: 'onDetailsPress'},
    onTransferPress: {action: 'onTransferPress'},
    onWithdrawPress: {action: 'onWithdrawPress'},
  },
} as ComponentMeta<typeof UserArea>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof UserArea> = args => (
  <UserArea {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
};
