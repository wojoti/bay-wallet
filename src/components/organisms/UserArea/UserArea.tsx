import {Container} from 'components/atoms';
import {BalanceBox, TransferAndWithdraw, UserBar} from 'components/molecules';
import {PropsWithChildren} from 'react';
import style from './UserArea.style';

export type UserAreaProps = PropsWithChildren<{
  onNotificationPress: () => void;
  onUserProfilePress: () => void;
  onDetailsPress: () => void;
  onTransferPress: () => void;
  onWithdrawPress: () => void;
  testId?: string;
}>;

const UserArea = ({
  onNotificationPress,
  onUserProfilePress,
  onDetailsPress,
  onTransferPress,
  onWithdrawPress,
  testId,
}: UserAreaProps) => (
  <Container
    flex={1}
    customStyle={style.contentWithRadius}
    align="center"
    testId={testId}>
    <Container customStyle={style.topContent}>
      <UserBar
        username={'David Kowalski'}
        onNotificationPress={onNotificationPress}
        onUserProfilePress={onUserProfilePress}
        testId="userarea-userbar"
      />
    </Container>
    <Container customStyle={style.balanceBoxSizer}>
      <BalanceBox
        onDetailsPress={onDetailsPress}
        amount={'45.500,12'}
        percent={'4.5'}
        balanceLabel={'Your main balance'}
        detailsLabel={'Details >'}
        testId="userarea-balancebox"
      />
    </Container>
    <Container customStyle={style.bottomContent}>
      <TransferAndWithdraw
        onTransferPress={onTransferPress}
        onWithdrawPress={onWithdrawPress}
        testId="userarea-transferandwithdraw"
      />
    </Container>
  </Container>
);

export default UserArea;
