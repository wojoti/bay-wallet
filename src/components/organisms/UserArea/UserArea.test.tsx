import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import UserArea, {UserAreaProps} from './UserArea';

const mockOnNotificationPress = jest.fn();
const mockOnUserProfilePress = jest.fn();
const mockOnDetailsPress = jest.fn();
const mockOnTransferPress = jest.fn();
const mockOnWithdrawPress = jest.fn();
const props: UserAreaProps = {
  onNotificationPress: mockOnNotificationPress,
  onUserProfilePress: mockOnUserProfilePress,
  onDetailsPress: mockOnDetailsPress,
  onTransferPress: mockOnTransferPress,
  onWithdrawPress: mockOnWithdrawPress,
  testId: 'test-userarea-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<UserArea {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render userarea', () => {
  render(<UserArea {...props} />);
  const userareaElement = screen.getByTestId('test-userarea-id');
  expect(userareaElement).toBeOnTheScreen();
});
