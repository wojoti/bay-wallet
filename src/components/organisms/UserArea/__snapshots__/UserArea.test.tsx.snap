// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "backgroundColor": "#363062",
        "borderBottomLeftRadius": 30,
        "borderBottomRightRadius": 30,
        "borderColor": "transparent",
        "borderWidth": 0,
      },
    ]
  }
  testID="test-userarea-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "marginTop": 60,
          "width": "85%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          undefined,
        ]
      }
      testID="userarea-userbar"
    >
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "flex": 2.6,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Poppins-Light",
              "fontSize": 14,
              "fontWeight": "400",
              "textAlign": "left",
            }
          }
          testID="userbar-welcome"
        >
          Good Morning
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Poppins-Bold",
              "fontSize": 24,
              "letterSpacing": undefined,
              "textAlign": "left",
            }
          }
          testID="userbar-username"
        >
          David Kowalski
        </Text>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderStyle": "solid",
                "justifyContent": "center",
                "opacity": 1,
              },
              false,
              undefined,
            ]
          }
          testID="userbar-notification"
        >
          <SvgMock
            height={26}
            width={24}
          />
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderStyle": "solid",
                "justifyContent": "center",
                "opacity": 1,
              },
              false,
              undefined,
            ]
          }
          testID="userbar-userprofile"
        >
          <SvgMock
            height={40}
            width={40}
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "marginTop": 35,
          "width": "93%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
      testID="userarea-balancebox"
    >
      <View
        style={
          [
            {
              "alignItems": "stretch",
              "flex": undefined,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": "100%",
            },
            {
              "backgroundColor": "#26204E",
              "borderRadius": 12,
              "height": 102,
              "position": "absolute",
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "alignItems": "stretch",
              "flex": undefined,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": "100%",
            },
            {
              "backgroundColor": "#F8F8FF",
              "borderRadius": 12,
              "height": 140,
              "width": "92%",
            },
          ]
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            [
              {
                "flex": 1,
              },
              {
                "flex": 1,
                "paddingLeft": 22,
                "paddingRight": 18,
                "paddingVertical": 14,
              },
            ]
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/img/wave.png",
              }
            }
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": undefined,
                  "width": undefined,
                },
                undefined,
              ]
            }
            testID="balancebox-background"
          />
          <View
            style={
              [
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#595959",
                  "fontFamily": "Poppins-Light",
                  "fontSize": 15,
                  "fontWeight": "400",
                  "textAlign": "left",
                }
              }
              testID="balancebox-label"
            >
              Your main balance
            </Text>
            <Text
              style={
                {
                  "color": "black",
                  "fontFamily": "Poppins-Bold",
                  "fontSize": 28,
                  "letterSpacing": undefined,
                  "textAlign": "left",
                }
              }
              testID="balancebox-amount"
            >
              $ 
              45.500,12
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "borderRadius": 8,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "height": 65,
                    "justifyContent": "center",
                    "opacity": 1,
                  },
                  false,
                  {
                    "borderWidth": 0,
                    "color": "transparent",
                    "height": 30,
                  },
                ]
              }
              testID="balancebox-button"
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 18,
                      "textAlign": "center",
                    },
                    {
                      "color": "#363062",
                      "fontSize": 15,
                    },
                  ]
                }
                testID="button-text"
              >
                Details &gt;
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "flex-start",
                    "flex": undefined,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                  {
                    "width": 90,
                  },
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#61C277",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 15,
                    "letterSpacing": undefined,
                    "textAlign": "left",
                  }
                }
                testID="balancebox-percent"
              >
                4.5
                 %
              </Text>
              <SvgMock
                height={20}
                testID="balancebox-wave"
                width={41}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "marginTop": 30,
          "width": "85%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
          undefined,
        ]
      }
      testID="userarea-transferandwithdraw"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderStyle": "solid",
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "#4E439E",
              "borderRadius": 14,
              "height": 60,
            },
          ]
        }
        testID="transfer-button"
      >
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": undefined,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
                "width": "100%",
              },
              {
                "width": 155,
              },
            ]
          }
        >
          <SvgMock
            height={24}
            testID="transfer-img"
            width={24}
          />
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "Poppins-Light",
                "fontSize": 15,
                "fontWeight": "400",
                "textAlign": "left",
              }
            }
            testID="transfer-label"
          >
            Transfer
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderStyle": "solid",
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "#4E439E",
              "borderRadius": 14,
              "height": 60,
            },
          ]
        }
        testID="withdraw-button"
      >
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": undefined,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
                "width": "100%",
              },
              {
                "width": 155,
              },
            ]
          }
        >
          <SvgMock
            height={24}
            testID="withdraw-img"
            width={24}
          />
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "Poppins-Light",
                "fontSize": 15,
                "fontWeight": "400",
                "textAlign": "left",
              }
            }
            testID="withdraw-label"
          >
            Withdraw
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
